wgpu-core/src/command/render.rs

```diff
@@ -346,14 +343,67 @@ impl State {
     }
 }
 
+/// Error encountered when performing a render pass.
+#[derive(Clone, Debug)]
+pub enum RenderPassError {
+    /// Attachment's sample count is not valid.
+    InvalidSampleCount,
+    /// All resolve targets must have a sample count of 1.
+    InvalidResolveTargetSampleCount,
+    /// Extent state must match extent from view.
+    ExtentStateMismatch,
+    /// Attempted to use a swap chain image as a depth/stencil attachment.
+    SwapChainImageAsDepthStencil,
+    /// Unable to clear non-present/read-only depth/stencil.
+    InvalidDepthStencilOps,
+    /// All attachments must have the same sample count.
+    SampleCountMismatch,
+    /// Texture view's swap chain must match swap chain in use.
+    SwapChainMismatch,
+    /// Setting `values_offset` to be `None` is only for internal use in render bundles.
+    InvalidValuesOffset,
+    /// The specified device feature is required for the given command, but is not enabled.
+    MissingDeviceFeature(&'static str),
+    /// Indirect draw command would overrun the bounds of the buffer.
+    IndirectBufferOverrun,
+    /// Can't pop debug group, because number of pushed debug groups is zero.
+    InvalidPopDebugGroup,
+    /// The render bundle output formats do not match render pass attachment formats.
+    IncompatibleRenderBundle,
+    /// An error encountered while encoding a command.
+    RenderCommandError(RenderCommandError),
+    /// An error encountered while handling a draw command.
+    DrawError(DrawError),
```

Comment:
similarly here and in other places, we don't need `Error` suffix in variants inside an `Error` kind of enum.
This could just be `Draw(DrawError)` and `Bind(BindError)`